spinn_front_end_common/utilities/helpful_functions.py

```diff
@@ -452,3 +455,43 @@ def convert_time_diff_to_total_milliseconds(sample):
     :return: total milliseconds
     """
     return (sample.total_seconds() * 1000.0) + (sample.microseconds / 1000.0)
+
+
+def determine_flow_states(executable_types, no_sync_changes):
```

Comment:
It would make more sense to include this information in the ExecutableType enum data structure as ExecutableType.get_start_states and ExecutableType.get_end_states.  If these return a set, this function can then just return the union of the sets.

Reply:
we agreed to not go down this route.

Reply:
But this means that if I add a type, I have to update this function...  That seems like an oversight...

Reply:
if you show me how to build it, ill add it in. as i noted it was a better process, just didnt feel had time to do it.

Reply:
fixed me thinks